### ISSUE TITLE
Something wrong with  F.cross_entropy(pred, gold, ignore_index=Constants.PAD, reduction="sum")''! !

### ISSUE THREAD

**Issue by charlesfufu** [jadore801120/attention-is-all-you-need-pytorch#61]
![image](https://user-images.githubusercontent.com/33311822/44980992-eae1d980-afa3-11e8-93d1-45504d4b5eb2.png)

Hi,could you tell me what's wrong with ''loss = F.cross_entropy(pred, gold, ignore_index=Constants.PAD, reduction="sum")''?

**Comment by charlesfufu** [comment]
Thanks for your answer。I have another question of which file has the function of predict。

**Comment by jadore801120** [comment]
Hi @charlesfufu , if you mean the prediction on test set, please refer to https://github.com/jadore801120/attention-is-all-you-need-pytorch/blob/master/transformer/Translator.py#L46-L166 , which handles the beam search on translation.
Thanks.
